lib/lib-search/src/main/java/org/sagebionetworks/search/SearchDaoImpl.java

```diff
@@ -130,18 +124,26 @@ public void createOrUpdateSearchDocument(List<Document> batch) throws ClientProt
 			for(Document document: documents){
 				prepareDocument(document);
 				serializedDocument = EntityFactory.createJSONStringForEntity(document);
-				// AwesomeSearch pukes on control characters. Some descriptions have
-				// control characters in them for some reason, in any case, just get rid
-				// of all control characters in the search document
-				String cleanedDocument = serializedDocument.replaceAll("\\p{Cc}", "");

```

Comment:
Would you not want to clean up the AwesomeSearch control characters (and escaped control characters) before submitting to markdown processing?  Could this markdown processing and escaping happen at the same time that the wiki source is added to the search document text?